docs/docsite/rst/vmware/scenario_remove_vm.rst

```diff
@@ -0,0 +1,127 @@
+.. _vmware_guest_remove_virtual_machine:
+
+*****************************************
+Remove an existing VMware virtual machine
+*****************************************
+
+.. contents:: Topics
+
+Introduction
+============
+
+This guide will show you how to utilize Ansible to remove an existing VMware virtual machine.
+
+Scenario Requirements
+=====================
+
+* Software
+
+    * Ansible 2.5 or later must be installed.
+
+    * The Python module ``Pyvmomi`` must be installed on the Ansible (or Target host if not executing against localhost).
+
+    * Installing the latest ``Pyvmomi`` via pip is recommended [as the OS packages are usually out of date and incompatible].
+
+* Hardware
+
+    * At least one standalone ESXi server or
+
+    * vCenter Server with at least one ESXi server
+
+* Access / Credentials
+
+    * Ansible (or the target server) must have network access to the either vCenter server or the ESXi server
+
+    * Username and Password for vCenter or ESXi server
+
+Assumptions
+===========
+
+- All variable names and VMware object names are case sensitive.
+- You need to use Python 2.7.9 version in order to use ``validate_certs`` option, as this version is capable of changing the SSL verification behaviours.
+
+Caveats
+=======
+
+- Hosts in the ESXi cluster must have access to the datastore that the template resides on.
+- ``vmware_guest`` module tries to mimick VMware Web UI and workflow, so the virtual machine must be in powered off state in order to remove it from the VMware inventory.
+
+.. warning::
+
+   The removal VMware virtual machine using ``vmware_guest`` module is destructive and can not be reverted, so it is recommended to take backup before proceeding.
+
+Example Description
+===================
+
+In this use case / example, user will be selecting a virtual machine using name for removal. The following Ansible playbook showcases the basic parameters that are needed for this.
+
+.. code-block:: yaml
+
+    ---
+    - name: Purge virtual machine
+      gather_facts: no
+      vars_files:
+        - vcenter_vars.yml
+      vars:
+        ansible_python_interpreter: "/usr/bin/env python3"
+      hosts: localhost
+      tasks:
+        - set_fact:
+            vm_name: "VM_0003"
+            datacenter: "DC1"
+
+        - name: Remove "{{ vm_name }}"
+          vmware_guest:
+            hostname: "{{ vcenter_server }}"
+            username: "{{ vcenter_user }}"
+            password: "{{ vcenter_pass }}"
+            validate_certs: no
+            cluster: "DC1_C1"
+            name: "{{ vm_name }}"
+            state: absent
+          delegate_to: localhost
+          register: facts
+
+
+Since Ansible utilizes the VMware API to perform actions, in this use case it will be connecting directly to the API from localhost.
```

Comment:
If a remove-VM play always runs on localhost, we don't need to tell the reader about installing things on the target host (see my comment on line 21).

Reply:
Pyvmomi can be installed on either Ansible Target node or other host from where you can connect to vCenter, so this will be required to clarify.

Reply:
Gotcha, thanks.